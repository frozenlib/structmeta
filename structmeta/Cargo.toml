[package]
name = "structmeta"
version = "0.1.0"
authors = ["frozenlib"]
license = "MIT OR Apache-2.0"
readme = "../README.md"
repository = "https://github.com/frozenlib/structmeta"
documentation = "https://docs.rs/structmeta/"
keywords = ["derive", "parse", "attribute", "syn", "totokens"]
categories = ["development-tools::procedural-macro-helpers"]
description = "Parse Rust's attribute arguments by defining a struct."
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
structmeta-derive = { version = "=0.1.0", path = "../structmeta-derive" }
proc-macro2 = "1.0.24"
syn = "1.0.65"
quote = "1.0.9"
