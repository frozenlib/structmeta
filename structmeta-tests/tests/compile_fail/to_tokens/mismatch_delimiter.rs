use structmeta::ToTokens;
use syn::LitStr;
#[derive(ToTokens)]
struct TestType {
    #[to_tokens("(")]
    braket_token: syn::token::Bracket,
    str: LitStr,
}

fn main() {}
