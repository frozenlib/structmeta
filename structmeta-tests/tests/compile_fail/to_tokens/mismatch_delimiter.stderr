error[E0308]: mismatched types
 --> tests/compile_fail/to_tokens/mismatch_delimiter.rs:6:5
  |
5 |     #[to_tokens("(")]
  |     - arguments to this function are incorrect
6 |     bracket_token: syn::token::Bracket,
  |     ^^^^^^^^^^^^^ expected `&Paren`, found `&Bracket`
  |
  = note: expected reference `&syn::token::Paren`
             found reference `&syn::token::Bracket`
note: method defined here
 --> $CARGO/syn-2.0.48/src/token.rs
  |
  | / define_delimiters! {
  | |     Brace         pub struct Brace        /// `{`&hellip;`}`
  | |     Bracket       pub struct Bracket      /// `[`&hellip;`]`
  | |     Parenthesis   pub struct Paren        /// `(`&hellip;`)`
  | | }
  | |_^
  = note: this error originates in the macro `define_delimiters` (in Nightly builds, run with -Z macro-backtrace for more info)
